### PR TITLE
Add beatmap tests + __repr__() method for objects

### PR DESCRIPTION
**Pull request by descensiontohell**
(needs a review)

- Gives each object in `osu/objects` a `__repr__()` method that utilizes `prettify()` function to display the class name of the object and some attributes specified as params
as
`User(username='nouvelle', id=6943941)`
`Beatmap(version="Bonzphyxia's Extra", beatmapset=Beatmapset(artist='USAO', title='Night sky', creator='sukiNathan'))`

- Adds auto tests for API methods based on pytest and the corresponding fixtures (only beatmap tests provided yet)

Running tests: 
1. Fill the credentials in `pytest_tests/constants.py`
2. Run `pytest pytest_tests`

I'm not sure if having pytest in `requirements.txt` will make users obtain that package on osu.py installation (it shouldn't happen since pytest is only used for dev purpose)

### CODE REVIEW

File: osu/objects/news.py [hunk 1]
```diff
@@ -70,3 +75,6 @@ class Navigation:
     def __init__(self, data):
         self.newer = NewsPost(data['newer']) if data["newer"] is not None else None
         self.older = NewsPost(data['older']) if data["older"] is not None else None
+
+    def __repr__(self):
```
Comment (Sheppsu) [inline osu/objects/news.py:79]: Is there a reason for leaving the fields blank here?

File: osu/objects/user.py [hunk 2]
```diff
@@ -547,3 +575,6 @@ def __init__(self, data, attr_type):
             print(f"WARNING: Unrecognized attr_type for CurrentUserAttributes: \"{attr_type}\"")
             for k, v in data.items():
                 setattr(self, k, v)
+
+    def __repr__(self):
```
Comment (Sheppsu) [inline osu/objects/user.py:583]: I think it would be better to only give the type attribute to prettify.

**Review by Sheppsu — CHANGES_REQUESTED** [review]

File: osu/objects/news.py (hunk 1 above)
Comment (descensiontohell) [reply osu/objects/news.py:79]: not really, solved in 844861766b61ccffad332d19c808cb9a558e82d2

File: osu/objects/user.py (hunk 2 above)
Comment (descensiontohell) [reply osu/objects/user.py:583]: solved in f227b2e034393567de3891b7351c8f3ab3462b35

**Comment by descensiontohell** [comment]
something broke again in tests, need to investigate

**Comment by descensiontohell** [comment]
apparently it's in new `parse_enum_args()` method so I recommend merging that and then fix the method based on tests

**Pull request merged**